packages/react-vapor/src/hoc/withDirty/withDirty.tsx

```diff
@@ -19,34 +19,38 @@ export interface IWithDirtyDispatchProps {
     toggleIsDirty: (isDirty: boolean) => void;
 }
 
-export interface IWithDirtyProps extends Partial<IWithDirtyStateProps>, Partial<IWithDirtyDispatchProps> {}
-
-export const withDirty = <T, R = any>(config: IWithDirty) => (
-    Component: React.ComponentType<IWithDirtyProps & T>
-): React.ComponentClass<IWithDirtyProps & T, R> => {
-    const mapStateToProps = (state: IReactVaporState): IWithDirtyStateProps => ({
-        isDirty: WithDirtySelectors.getIsDirty(state, {id: config.id}),
+export interface IWithDirtyProps
+    extends Partial<IWithDirty>,
+        Partial<IWithDirtyStateProps>,
+        Partial<IWithDirtyDispatchProps> {}
+
+export const withDirty = <T, R = any>(config: Partial<IWithDirty> = {}) => (
+    Component: React.ComponentType<Partial<IWithDirtyProps> & T>
+): React.ComponentClass<Partial<IWithDirtyProps> & T, R> => {
+    const mapStateToProps = (state: IReactVaporState, ownProps: IWithDirty): IWithDirtyStateProps => ({
+        isDirty: WithDirtySelectors.getIsDirty(state, {id: config.id || ownProps.id}),
     });
 
-    const mapDispatchToProps = (dispatch: IDispatch): IWithDirtyDispatchProps => ({
-        toggleIsDirty: (isDirty: boolean) => dispatch(WithDirtyActions.toggle(config.id, isDirty)),
+    const mapDispatchToProps = (dispatch: IDispatch, ownProps: IWithDirty): IWithDirtyDispatchProps => ({
+        toggleIsDirty: (isDirty: boolean) => dispatch(WithDirtyActions.toggle(config.id || ownProps.id, isDirty)),
     });
 
     @ReduxConnect(mapStateToProps, mapDispatchToProps)
     class ComponentWithDirty extends React.PureComponent<Partial<IWithDirtyProps> & T, R> {
         componentDidMount() {
-            this.props.toggleIsDirty(config.isDirty);
+            this.props.toggleIsDirty(config?.isDirty || this.props?.isDirty);

```

Comment:
Same here, I think we should invert the order if we want to get rid of the config someday to only use the props.